src/main/java/Listener/CountingEnforcement.java

```diff
@@ -39,24 +42,30 @@ private void countCheck(TextChannel channel, Message message, MessageAuthor auth
             return;
         }
 
-        Integer input;
+        Integer previousNum;
         User user;
-        if (channel.getMessages(2).get().getOldestMessage().get().getAuthor().getIdAsString().equals(BotInfo.getBotId())) {
-            input = Integer.parseInt(channel.getMessages(3).get().getOldestMessage().get().getContent().replaceAll("\\*", "")) + 1;
-            user = channel.getMessages(3).get().getOldestMessage().get().getAuthor().asUser().get();
-        } else {
-            input = Integer.parseInt(channel.getMessages(2).get().getOldestMessage().get().getContent().replaceAll("\\*", "")) + 1;
-            user = channel.getMessages(2).get().getOldestMessage().get().getAuthor().asUser().get();
-        }
+        try {
+            if (channel.getMessages(2).get().getOldestMessage().get().getAuthor().getIdAsString().equals(BotInfo.getBotId())) {
+                previousNum = parseIntFromMessage(channel.getMessages(3).get().getOldestMessage().get().getContent().replaceAll("\\*", "")); 
+                user = channel.getMessages(3).get().getOldestMessage().get().getAuthor().asUser().get();
+            } else {
+                previousNum = parseIntFromMessage(channel.getMessages(2).get().getOldestMessage().get().getContent().replaceAll("\\*", ""));
+                user = channel.getMessages(2).get().getOldestMessage().get().getAuthor().asUser().get();
+            }
 
-        if (!input.equals(Integer.parseInt(msg))) {
-            channel.sendMessage("<@" + author.getIdAsString() + ">" + ", please input the correct next number");
-            Thread.sleep(1250);
-            message.delete();
-        } else if (author.getIdAsString().equals(user.getIdAsString())) {
-            channel.sendMessage("<@" + author.getIdAsString() + ">" + ", double counting isn't allowed. Please wait until someone else has counted");
-            Thread.sleep(1250);
-            message.delete();
+            if (!previousNum.equals(msgNum - 1)) {
+                channel.sendMessage("<@" + author.getIdAsString() + ">" + ", please input the correct next number");
+                Thread.sleep(1250);
+                message.delete();
+            } else if (author.getIdAsString().equals(user.getIdAsString())) {
+                channel.sendMessage("<@" + author.getIdAsString() + ">" + ", double counting isn't allowed. Please wait until someone else has counted");
+                Thread.sleep(1250);
+                message.delete();
+            }
```

Comment:
This check has nothing to do with numbers and could be outside the try-catch

Reply:
that would be better because then it requires teamwork to exploit

Reply:
fixed by https://github.com/IKanraI/Gaiza/pull/138/commits/2278fecb3bd71fc836afcc3bee8a65d4e22e1a46